flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/manifest/ManifestListWriter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.manifest;
+
+import org.apache.flink.api.common.serialization.BulkWriter;
+import org.apache.flink.core.fs.FSDataOutputStream;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Writer for a manifest list. This file includes several {@link ManifestFileMeta}, representing all
+ * data of the whole table at the corresponding snapshot.
+ */
+public class ManifestListWriter {
+
+    private final ManifestFileMetaSerializer serializer;
+    private final BulkWriter.Factory<RowData> writerFactory;
+
+    public ManifestListWriter(
+            RowType partitionType,
+            Function<RowType, BulkWriter.Factory<RowData>> writerFactoryCreator) {
+        this.serializer = new ManifestFileMetaSerializer(partitionType);
+        this.writerFactory = writerFactoryCreator.apply(ManifestFileMeta.schema(partitionType));
+    }
+
+    public void write(List<ManifestFileMeta> metas, Path path) throws IOException {
+        Preconditions.checkArgument(
```

Comment:
We can have a `FileUtils.writeListToFile`?

Reply:
We can't, at least currently. Most of the writers have some special logic (for example collecting the stats, calculating the number of added files) so it is hard to extract a utility method.